docs/start/framework/react/reading-writing-file.md

```diff
@@ -0,0 +1,557 @@
+
+- Basic knowledge of React and TypeScript.
+- Node.js and `pnpm` installed on your machine
+
+## Nice to know
+
+- [Server Side Rendering (SSR)](https://tanstack.com/router/latest/docs/framework/react/guide/ssr)
+- [TanStack Router concepts](https://tanstack.com/router/latest/docs/framework/react/routing/routing-concepts)
+- [React Query concepts](https://tanstack.com/query/latest/docs/framework/react/overview)
+
+## Setting up a TanStack Start Project
+
+First, let's create a new TanStack Start project:
+
+```bash
+pnpx create-start-app devjokes
+cd devjokes
+```
+
+When this script runs, it will ask you a few setup questions. You can either pick choices that work for you or just press enter to accept the defaults.
+
+Optionally, you can pass in a `--add-on` flag to get options such as Shadcn, Clerk, Convex, TanStack Query, etc.
+
+Once setup is complete, install dependencies and start the development server:
+
+```bash
+pnpm i
+pnpm dev
+```
+
+For this project, we'll need a few additional packages:
+
+```bash
+# Install uuid for generating unique IDs
+pnpm add uuid
+pnpm add -D @types/uuid
+```
+
+## Understanding the project structure
+
+At this point, the project structure should look like this -
+
+```
+/devjokes
+├── src/
+│   ├── routes/
+│   │   ├── __root.tsx                    # Root layout
+│   │   ├── index.tsx                     # Home page
+│   │   ├── demo.start.server-funcs.tsx   # Demo server functions
+│   │   └── demo.start.api-request.tsx    # Demo API request
+│   ├── api/                              # API endpoints
+│   ├── components/                       # React components
+│   ├── api.ts                            # API handler.
+│   ├── client.tsx                        # Client entry point
+│   ├── router.tsx                        # Router configuration
+│   ├── routeTree.gen.ts                  # Generated route tree
+│   ├── ssr.tsx                           # Server-side rendering
+│   └── styles.css                        # Global styles
+├── public/                               # Static assets
+├── app.config.ts                         # TanStack Start configuration
+├── package.json                          # Project dependencies
+└── tsconfig.json                         # TypeScript configuration
+```
+
+This structure might seem overwhelming at first, but here are the key files you need to focus on:
+
+1. `router.tsx` - Sets up routing for your application
+2. `src/routes/__root.tsx` - The root layout component where you can add global styles and components
+3. `src/routes/index.tsx` - Your home page
+4. `client.tsx` - Client-side entry point
+5. `ssr.tsx` - Handles server-side rendering
+
+Once your project is set up, you can access your app at `localhost:3000`. You should see the default TanStack Start welcome page.
+
+At this point, your app will look like this -
+
+![TanStack Start Welcome Page After Setup](https://res.cloudinary.com/dubc3wnbv/image/upload/v1746312482/Photo-1_lpechn.png)
```

Comment:
Could we inline the images in the repo?

See, `docs/router/framework/react/guide/custom-search-param-serialization.md`.

This should have it work where the website will automatically choose the image based on the selected branch.